src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -22873,6 +23317,105 @@ class Derived2 : Base, I<(int a, int b)>
                 );
         }
 
+        [Fact]
+        [WorkItem(14841, "https://github.com/dotnet/roslyn/issues/14841")]
+        public void ExplicitBaseImplementationNotConsideredImplementationForInterfaceWithDifferentTupleNames_02()
```

Comment:
>ExplicitBaseImplementationNotConsideredImplementationForInterfaceWithDifferentTupleNames_02 [](start = 20, length = 91)

Looks like a new error (compat issue). It would be good to dig up the latest LDM notes on the topic. If I recall, the previous behavior was intentional.

Reply:
There is no intent to change language rules here. There was a bug that compiler picked ```public (int notA, int notB) M() { return (1, 2); }``` as an implicit implementation for the interface, but that is incorrect because for CLR tuple names do not matter and explicit implementation takes priority. The behavior should be similar to one in
ExplicitBaseImplementationNotConsideredImplementationForInterfaceWithDifferentTupleNames_01

---
In reply to: [244438573](https://github.com/dotnet/roslyn/pull/32041#discussion_r244438573) [](ancestors = 244438573)